src/main/java/fr/esgi/al5/tayarim/services/LogementService.java

```diff
@@ -0,0 +1,291 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import fr.esgi.al5.tayarim.dto.logement.LogementCreationDto;
+import fr.esgi.al5.tayarim.dto.logement.LogementDto;
+import fr.esgi.al5.tayarim.dto.logement.LogementUpdateDto;
+import fr.esgi.al5.tayarim.entities.Logement;
+import fr.esgi.al5.tayarim.entities.Proprietaire;
+import fr.esgi.al5.tayarim.exceptions.LogementInvalidCreationBody;
+import fr.esgi.al5.tayarim.exceptions.LogementInvalidUpdateBody;
+import fr.esgi.al5.tayarim.exceptions.LogementNotFoundException;
+import fr.esgi.al5.tayarim.exceptions.ProprietaireInvalidUpdateBody;
+import fr.esgi.al5.tayarim.exceptions.ProprietaireNotFoundException;
+import fr.esgi.al5.tayarim.mappers.LogementMapper;
+import fr.esgi.al5.tayarim.repositories.LogementRepository;
+import fr.esgi.al5.tayarim.repositories.ProprietaireRepository;
+import java.time.LocalTime;
+import java.util.List;
+import java.util.Optional;
+import lombok.NonNull;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.client.RestTemplate;
+
+
+/**
+ * Classe de service gérant les logements.
+ */
+@Service
+@Transactional(readOnly = true)
+public class LogementService {
+
+  private final LogementRepository logementRepository;
+  private final ProprietaireRepository proprietaireRepository;
+
+
+  /**
+   * Constructeur pour le service de logement.
+   *
+   * @param logementRepository     Le repository des logements.
+   * @param proprietaireRepository Le repository des propriétaires.
+   */
+  public LogementService(LogementRepository logementRepository,
+      ProprietaireRepository proprietaireRepository) {
+    this.logementRepository = logementRepository;
+    this.proprietaireRepository = proprietaireRepository;
+  }
+
+  /**
+   * Tente de créer un logement.
+   *
+   * @param logementCreationDto Le dto de création de logement.
+   * @return {@link LogementDto}
+   */
+  @Transactional
+  public LogementDto createLogement(@NonNull LogementCreationDto logementCreationDto) {
+
+    Optional<Proprietaire> optionalProprietaire = proprietaireRepository.findById(
+        logementCreationDto.getIdProprietaire());
+
+    if (optionalProprietaire.isEmpty()) {
+      throw new ProprietaireNotFoundException();
+    }
+
+    Proprietaire proprietaire = optionalProprietaire.get();
+
+    if(
+      Boolean.parseBoolean(logementCreationDto.getIsLouable()) && (
+        logementCreationDto.getNombresDeChambres() == null
+        || logementCreationDto.getNombresDeLits() == null
+        || logementCreationDto.getNombresSallesDeBains() == null
+        || logementCreationDto.getCapaciteMaxPersonne() == null
+        || logementCreationDto.getNombresNuitsMin() == null
+        || (logementCreationDto.getDefaultCheckIn() == null || logementCreationDto.getDefaultCheckIn().isBlank())
+        || (logementCreationDto.getDefaultCheckOut() == null || logementCreationDto.getDefaultCheckOut().isBlank())
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 115).